### PR TITLE
fix(Dew): fix dew instance resource lint error

### PR DESCRIPTION
**Pull request by wangyuancheng01**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(Using lint to scan issues with dev modules)*
fixes #dev

**Special notes for your reviewer**:

**Release note**:

```==> Checking for running environment...

==> Applying patch...

==> Checking for code complexity...
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Language                              Files     Lines   Blanks  Comments     Code Complexity Complexity/Lines
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Go                                        2       828      111        18      699        110            30.45
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
~s/dsc/resource_huaweicloud_dsc_instance.go       545       70        10      465         78            16.77
~/dsc/resource_huaweicloud_dsc_asset_obs.go       283       41         8      234         32            13.68
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Total                                     2       828      111        18      699        110            30.45
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Processed 25685 bytes, 0.026 megabytes (SI)
─────────────────────────────────────────────────────────────────────────────────────────────────────────────

the TOP10 most complex functions:
11 dsc resourceDscInstanceCreate huaweicloud/services/dsc/resource_huaweicloud_dsc_instance.go:126:1
7 dsc buildCreateDscInstanceRequestBodyProductInfos huaweicloud/services/dsc/resource_huaweicloud_dsc_instance.go:244:1
7 dsc resourceAssetObsCreate huaweicloud/services/dsc/resource_huaweicloud_dsc_asset_obs.go:62:1
5 dsc waitForBmsInstanceDelete huaweicloud/services/dsc/resource_huaweicloud_dsc_instance.go:503:1
5 dsc resourceDscInstanceRead huaweicloud/services/dsc/resource_huaweicloud_dsc_instance.go:419:1
5 dsc addProductIdToProductInfo huaweicloud/services/dsc/resource_huaweicloud_dsc_instance.go:308:1
5 dsc buildCreateDscInstanceBodyParams huaweicloud/services/dsc/resource_huaweicloud_dsc_instance.go:210:1
4 dsc buildProductInfo huaweicloud/services/dsc/resource_huaweicloud_dsc_instance.go:384:1
4 dsc getOrderProducts huaweicloud/services/dsc/resource_huaweicloud_dsc_instance.go:344:1
4 dsc resourceAssetObsUpdate huaweicloud/services/dsc/resource_huaweicloud_dsc_asset_obs.go:209:1
Average: 3.68

==> Checking for golangci-lint...

==> Checking for Nolint directives...

==> Checking for TF features in dsc...

==> Checking for misspell in dsc...

==> Checking for code complexity in ./huaweicloud/services/acceptance/dsc...
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Language                              Files     Lines   Blanks  Comments     Code Complexity Complexity/Lines
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Go                                        2       236       28         2      206         16            15.65
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
~resource_huaweicloud_dsc_asset_obs_test.go       127       14         1      112          8             7.14
~/resource_huaweicloud_dsc_instance_test.go       109       14         1       94          8             8.51
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Total                                     2       236       28         2      206         16            15.65
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Processed 7201 bytes, 0.007 megabytes (SI)
─────────────────────────────────────────────────────────────────────────────────────────────────────────────

the TOP5 most complex functions:
5 dsc getDscInstanceResourceFunc huaweicloud/services/acceptance/dsc/resource_huaweicloud_dsc_instance_test.go:18:1
5 dsc getAssetObsResourceFunc huaweicloud/services/acceptance/dsc/resource_huaweicloud_dsc_asset_obs_test.go:19:1
1 dsc testDscInstance_basic huaweicloud/services/acceptance/dsc/resource_huaweicloud_dsc_instance_test.go:97:1
1 dsc TestAccDscInstance_basic huaweicloud/services/acceptance/dsc/resource_huaweicloud_dsc_instance_test.go:57:1
1 dsc testAssetObs_basic huaweicloud/services/acceptance/dsc/resource_huaweicloud_dsc_asset_obs_test.go:103:1
Average: 2.33

==> Checking for golangci-lint in ./huaweicloud/services/acceptance/dsc...

==> Checking for Nolint directives in ./huaweicloud/services/acceptance/dsc...

==> Cleanup patch...

Check Completed!

```

## PR Checklist

* [x] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
root@ecs-wangyuancheng:/home/huawei/go/src/github.com/huaweicloud/terraform-provider-huaweicloud# make testacc TEST="./huaweicloud/services/acceptance/dew" TESTARGS="-run TestAccDewCsmsSecretVersion_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dew -v -run TestAccDewCsmsSecretVersion_basic -timeout 360m -parallel 4
=== RUN   TestAccDewCsmsSecretVersion_basic
=== PAUSE TestAccDewCsmsSecretVersion_basic
=== CONT  TestAccDewCsmsSecretVersion_basic
--- PASS: TestAccDewCsmsSecretVersion_basic (26.68s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dew       26.713s
root@ecs-wangyuancheng:/home/huawei/go/src/github.com/huaweicloud/terraform-provider-huaweicloud# make testacc TEST="./huaweicloud/services/acceptance/dew" TESTARGS="-run TestAccDataKpsKeypairs_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dew -v -run TestAccDataKpsKeypairs_basic -timeout 360m -parallel 4
=== RUN   TestAccDataKpsKeypairs_basic
=== PAUSE TestAccDataKpsKeypairs_basic
=== CONT  TestAccDataKpsKeypairs_basic
--- PASS: TestAccDataKpsKeypairs_basic (15.93s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dew       15.982s
root@ecs-wangyuancheng:/home/huawei/go/src/github.com/huaweicloud/terraform-provider-huaweicloud# make testacc TEST="./huaweicloud/services/acceptance/dew" TESTARGS="-run TestAccDewCsmsSecret_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dew -v -run TestAccDewCsmsSecret_basic -timeout 360m -parallel 4
=== RUN   TestAccDewCsmsSecret_basic
=== PAUSE TestAccDewCsmsSecret_basic
=== CONT  TestAccDewCsmsSecret_basic
--- PASS: TestAccDewCsmsSecret_basic (23.18s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dew       23.218s
root@ecs-wangyuancheng:/home/huawei/go/src/github.com/huaweicloud/terraform-provider-huaweicloud# make testacc TEST="./huaweicloud/services/acceptance/dew" TESTARGS="-run TestAccKmsKey_Basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dew -v -run TestAccKmsKey_Basic -timeout 360m -parallel 4
=== RUN   TestAccKmsKey_Basic
    acceptance.go:695: This environment does not support KMS tests
--- SKIP: TestAccKmsKey_Basic (0.00s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dew       0.037s

```

### CODE REVIEW

File: huaweicloud/services/dew/data_source_huaweicloud_kms_key.go [hunk 1]
```diff
@@ -99,30 +99,30 @@ func DataSourceKmsKey() *schema.Resource {
 }
 
 func dataSourceKmsKeyRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	config := meta.(*config.Config)
-	region := config.GetRegion(d)
-	kmsKeyV1Client, err := config.KmsKeyV1Client(region)
+	configuration := meta.(*config.Config)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/dew/data_source_huaweicloud_kms_key.go]: better to use cfg
```suggestion
	cfg := meta.(*config.Config)
```

File: huaweicloud/services/dew/resource_huaweicloud_csms_secret.go [hunk 2]
```diff
@@ -308,7 +308,7 @@ func resourceCsmsSecretImport(ctx context.Context, d *schema.ResourceData, _ int
 	return []*schema.ResourceData{d}, nil
 }
 
-func parseID(id string) (string, string) {
+func parseID(id string) (str1 string, str2 string) {
 	parts := strings.SplitN(id, "/", 2)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/dew/resource_huaweicloud_csms_secret.go]: SplitN should not be used
```suggestion
	parts := strings.Split(id, "/")
```

File: huaweicloud/services/dew/resource_huaweicloud_csms_secret.go [hunk 3]
```diff
@@ -265,40 +265,40 @@ func resourceCsmsSecretUpdate(ctx context.Context, d *schema.ResourceData, meta
...
 	return nil
 }
 
-func resourceCsmsSecretImport(ctx context.Context, d *schema.ResourceData, _ interface{}) ([]*schema.ResourceData,
+func resourceCsmsSecretImport(_ context.Context, d *schema.ResourceData, _ interface{}) ([]*schema.ResourceData,
 	error) {
 	id, name := parseID(d.Id())
 	if id == "" {
-		err := fmtp.Errorf("Invalid format specified for the ID of CSMS secret. " +
+		err := fmt.Errorf("Invalid format specified for the ID of CSMS secret. " +
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/dew/resource_huaweicloud_csms_secret.go]: error message should start with lower case, please check other messages
```suggestion
		err := fmt.Errorf("invalid format specified for the ID of CSMS secret. " +
```

File: huaweicloud/services/dew/resource_huaweicloud_kms_key.go [hunk 4]
```diff
@@ -135,9 +135,9 @@ func resourceKmsKeyValidation(d *schema.ResourceData) error {
 }
 
 func ResourceKmsKeyCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	config := meta.(*config.Config)
-	region := config.GetRegion(d)
-	kmsKeyV1Client, err := config.KmsKeyV1Client(region)
+	configuration := meta.(*config.Config)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/dew/resource_huaweicloud_kms_key.go]:
```suggestion
	cfg := meta.(*config.Config)
```

File: huaweicloud/services/dew/resource_huaweicloud_csms_secret.go [hunk 5]
```diff
@@ -240,11 +240,11 @@ func resourceCsmsSecretUpdate(ctx context.Context, d *schema.ResourceData, meta
...
 			Description: &desc,
 		}
-		logp.Printf("[DEBUG] The option to update the basic information of the CSMS secret is: %#v", opts)
+		log.Printf("[DEBUG] The option to update the basic information of the CSMS secret is: %#v", opts)
 
 		_, err = secrets.Update(client, name, opts)
 		if err != nil {
-			e := fmtp.Errorf("failed to update the base-info of CSMS secret: %s", err)
+			e := fmt.Errorf("failed to update the base-info of CSMS secret: %s", err)
 			mErr = multierror.Append(mErr, e)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/dew/resource_huaweicloud_csms_secret.go]: directly return the error, don't need to use multierror

File: huaweicloud/services/dew/resource_huaweicloud_csms_secret.go [hunk 6]
```diff
@@ -256,49 +255,47 @@ func resourceCsmsSecretUpdate(ctx context.Context, d *schema.ResourceData, meta
...
-			e := fmtp.Errorf("failed to update CSMS secret tags: %s", err)
-			mErr = multierror.Append(mErr, e)
+			return diag.Errorf("failed to update CSMS secret tags: %s", err)
 		}
 	}
 
 	if mErr.ErrorOrNil() != nil {
-		return fmtp.DiagErrorf("failed to update CSMS secret: %s", mErr)
+		return diag.Errorf("failed to update CSMS secret: %s", mErr)
 	}
```
Comment (ShiChangkuo) [inline huaweicloud/services/dew/resource_huaweicloud_csms_secret.go]: mErr can be deleted as you just return when an error raised during update.

File: huaweicloud/services/dew/data_source_huaweicloud_kms_key.go (hunk 1 above)
Comment (wangyuancheng01) [reply huaweicloud/services/dew/data_source_huaweicloud_kms_key.go]: fixed

File: huaweicloud/services/dew/resource_huaweicloud_csms_secret.go (hunk 2 above)
Comment (wangyuancheng01) [reply huaweicloud/services/dew/resource_huaweicloud_csms_secret.go]: fixed

File: huaweicloud/services/dew/resource_huaweicloud_csms_secret.go (hunk 3 above)
Comment (wangyuancheng01) [reply huaweicloud/services/dew/resource_huaweicloud_csms_secret.go]: fixed

File: huaweicloud/services/dew/resource_huaweicloud_kms_key.go (hunk 4 above)
Comment (wangyuancheng01) [reply huaweicloud/services/dew/resource_huaweicloud_kms_key.go]: fixed

File: huaweicloud/services/dew/resource_huaweicloud_csms_secret.go (hunk 5 above)
Comment (wangyuancheng01) [reply huaweicloud/services/dew/resource_huaweicloud_csms_secret.go]: fixed

File: huaweicloud/services/dew/resource_huaweicloud_csms_secret.go (hunk 6 above)
Comment (wangyuancheng01) [reply huaweicloud/services/dew/resource_huaweicloud_csms_secret.go]: done

**Pull request merged**